app/code/Magento/Integration/Model/CustomerTokenService.php

```diff
@@ -99,7 +109,11 @@ public function createCustomerAccessToken($username, $password)
                 )
             );
         }
-        $this->eventManager->dispatch('customer_login', ['customer' => $customerDataObject]);
+
+        $customerModel = $this->customerRegistry->retrieve($customerDataObject->getId());
+        $this->eventManager->dispatch('customer_login', ['customer' => $customerModel]);
+        $this->eventManager->dispatch('customer_data_object_login', ['customer' => $customerDataObject]);
```

Comment:
Hello @stefliekens ,
thank you for your contribution.

I have a question, this event was already fired by: 

$customerDataObject = $this->accountManagement->authenticate($username, $password);

at row \Magento\Customer\Model\AccountManagement::authenticate:622

Can you check if it will be fired two times?

Reply:
Hi @danielrussob, 
You are right, but the event `customer_data_object_login` is also fired twice in a regular login via the storefront. 
Check the `execute` method in `\Magento\Customer\Controller\Account\LoginPost`:
```
$customer = $this->customerAccountManagement->authenticate($login['username'], $login['password']);
$this->session->setCustomerDataAsLoggedIn($customer);
```
During `authenticate()` and `setCustomerDataAsLoggedIn()` you can see the same event is fired. 

Any suggestions or thoughts?

Reply:
Hello @stefliekens ,
I confirm that customer_data_object_login is also fired twice in a regular login via the storefront, I don't know why..

Maybe it's better talk to QA team, I mention @sdzhepa to start a discussion about it